lib/dal/src/pkg/import.rs

```diff
@@ -216,36 +214,64 @@ async fn import_change_set(
                     Thing::Func(func.to_owned()),
                 );
 
-                None
-            } else {
-                Some(
-                    import_func(
-                        ctx,
-                        func_spec,
-                        installed_module.clone(),
-                        thing_map,
-                        options.create_unlocked,
-                    )
-                    .await?,
-                )
-            };
+                continue;
+            }
 
-            if let Some(func) = func {
-                thing_map.insert(
-                    func_spec.unique_id().to_owned(),
-                    Thing::Func(func.to_owned()),
-                );
+            let func_spec_data = func_spec
+                .data()
+                .ok_or(PkgError::DataNotFound(func_spec.name().into()))?;
 
-                if let Some(module) = installed_module.clone() {
-                    module.create_association(ctx, func.id.into()).await?;
-                }
+            // If this func is a transformation, we need to see if it updates an existing func before creating it
+            if func_spec_data.is_transformation() {
```

Comment:
Doesn't this violate `is_locked`? Schema variants get a new ID when you unlock them and update, and so do funcs. I'm worried violating this principle will make it harder to reason about the system and have have downstream implications on the rest of the graph, where we very probably already make assumptions that locked FuncIds never change ... for example, when you update a locked transformation func, do you re-run DVU for everything that uses it?

If we don't *need* to support updating existing transformation funcs, I suggest we leave it for now--we need to think it through more comprehensively before we support it. We can still update transformation funcs, we just need to include the new Func (with new FuncId) in an updated Region/AWS Credential package.